src/tests/unit/substitutionsTest.ts

```diff
@@ -26,29 +29,219 @@ const displayMap = {
+                    body: "Hello world!",
+                },
+                sender: "@alice:localhost",
+            }, fakeMain, "");
+            expect(res).to.deep.equal({
+                link_names: true,
+                text: "Hello world!",
+                username: "@alice:localhost",
+            });
         });
-    });
-    describe("makeFirstWordMap", () => {
-        const expectedFirstWords = {
-            Alice: [{Alice: "U2222"}],
-            Bob: [{Bob: "U1111"},
-            {"Bob Charlie": "U3333"}],
-        };
-        it("should make the first word map", () => {
-            const firstWords = subsitutions.makeFirstWordMap(displayMap);
-            expect(firstWords).to.deep.equal(expectedFirstWords);
+        it ("should allow a simple m.text message", async () => {
+            const res = await subsitutions.matrixToSlack({
+                content: {
+                    body: "Hello world!",
+                    msgtype: "m.text",
+                },
+                sender: "@alice:localhost",
+            }, fakeMain, "");
+            expect(res).to.deep.equal({
+                link_names: true,
+                text: "Hello world!",
+                username: "@alice:localhost",
+            });
+        });
+        it ("should allow a simple message with any other type", async () => {
+            const res = await subsitutions.matrixToSlack({
+                content: {
+                    body: "Hello world!",
+                    msgtype: "org.matrix.fake.message.type",
+                },
+                sender: "@alice:localhost",
+            }, fakeMain, "");
+            expect(res).to.deep.equal({
+                link_names: true,
+                text: "Hello world!",
+                username: "@alice:localhost",
+            });
+        });
+        it ("should handle m.emote", async () => {
+            const res = await subsitutions.matrixToSlack({
+                content: {
+                    body: "This is not sarcasm",
+                    msgtype: "m.emote",
+                },
+                sender: "@alice:localhost",
+            }, fakeMain, "");
+            expect(res).to.deep.equal({
+                link_names: true,
+                text: "_This is not sarcasm_",
+                username: "@alice:localhost",
+            });
+        });
+        it ("should replace <, > and & with HTML codes", async () => {
+            const res = await subsitutions.matrixToSlack({
+                content: {
+                    body: "Let's talk about <html> and the & character",
+                },
+                sender: "@alice:localhost",
+            }, fakeMain, "");
+            expect(res).to.deep.equal({
+                link_names: true,
+                text: "Let's talk about &lt;html&gt; and the &amp; character",
+                username: "@alice:localhost",
+            });
+        });
+        it ("should replace @channel with @room", async () => {
```

Comment:
Also `@here` and `@everyone`?

Reply:
The title is wrong, should be replace @\room with @\channel. And Matrix only supports @\room